src/software/new_geom/circle.cpp

```diff
@@ -0,0 +1,53 @@
+#include "software/new_geom/circle.h"
+
+Circle::Circle() : origin_(Point()), radius_(0) {}
+
+Circle::Circle(const Point &origin, double radius) : origin_(origin)
+{
+    setRadius(radius);
+}
+
+void Circle::setOrigin(const Point &o)
+{
+    origin_ = o;
+}
+
+Point Circle::getOrigin() const
+{
+    return Point(origin_);
```

Comment:
I think returning a copy here is redundant, it should already be returning a copy

Reply:
Fixed